javascripts/discourse/initializers/initialize-for-header-icon-links.gjs

```diff
@@ -38,24 +32,23 @@ export default {
           const rel = target ? "noopener" : "";
           const isLastLink =
             link === links[links.length - 1] ? ".last-custom-icon" : "";
-          const selector = `li.custom-header-icon-link.${className}.${viewClass}${isLastLink}`;
 
-          api.decorateWidget("header-icons:before", (helper) => {
-            return helper.h(selector, [
-              helper.h(
-                "a.icon.btn-flat",
-                {
-                  href,
-                  title,
-                  target,
-                  attributes: {
-                    rel,
-                  },
-                },
-                icon
-              ),
-            ]);
-          });
+          const iconComponent = <template>
+            <li class="custom-header-icon-link {{className}} {{viewClass}} {{isLastLink}}">
+              <a class="icon btn-flat"
+              href="{{href}}"
+              title="{{title}}"
+              target="{{target}}"
+              rel="{{rel}}"
+              >
+                {{icon}}
+              </a>
+            </li>
+          </template>
+
+          const beforeIcon = ['chat', 'search', 'hamburger', 'user-menu']
```

Comment:
[Plugin API docs](https://github.com/discourse/discourse/blob/0c827980e6689df333143a43641924f7f8e38d02/app/assets/javascripts/discourse/app/lib/plugin-api.gjs#L1820) say that new icons are by default added to the left of existing icons, but "chat" is still placed furthest left unless a `before` is provided.
I imagine the chat button isn't always there, so should we provide this array as a series of fall-back "first icon" names even though that seems to be an undocumented feature, or should we just use "chat", or something else?

dag.js does say an array of keys is supported, so maybe we're good!

Reply:
The docs should probably say 'to the left of core icons'. Chat is a plugin, so it uses this same `api.headerIcons` system we're using here.

I like this solution of adding multiple 'before' items in an array 👌